File: ComponentModule/library-common/src/main/java/com/lee/library/base/BaseApplication.java
package com.lee.library.base;

import android.app.Application;
import android.util.Log;

import com.lee.library.utils.Constants;

/**
 * @author jv.lee
 * @date 2019/7/10.
 * @description
 */
public class BaseApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        Log.e(Constants.TAG, "common/BaseApplication");
    }
}

File: KotlinLibrary/library/src/main/java/com/lee/library/base/BaseApplication.kt
package com.lee.library.base

import android.content.Context
import androidx.multidex.MultiDexApplication

/**
 * @author jv.lee
 * @date 2020/3/20
 * @description
 */
abstract class BaseApplication : MultiDexApplication() {

    companion object {
        private lateinit var sContext: Context

        fun getContext(): Context {
            return sContext
        }
    }

    override fun onCreate() {
        super.onCreate()
        sContext = this
        init()
    }

    override fun onTerminate() {
        super.onTerminate()
        unInit()
    }

    protected abstract fun init()
    protected abstract fun unInit()

}